tasks/eva_2406/gather_release_species.py

```diff
@@ -0,0 +1,357 @@
+        query_create_table = (
+            'CREATE TABLE IF NOT EXISTS remapping_progress '
+            '(source TEXT, taxid INTEGER, scientific_name TEXT, assembly_accession TEXT, number_of_study INTEGER NOT NULL,'
+            'number_submitted_variants BIGINT NOT NULL, release_number INTEGER, `target_assembly_accession` TEXT, '
+            'report_time TIMESTAMP DEFAULT NOW(), progress_status TEXT, start_time TIMESTAMP, '
+            'completion_time TIMESTAMP, '
+            'primary key(source, taxid, assembly_accession, release_number))'
+        )
+    execute_query(metadata_connection_handle, query_create_table)
+
+
+def insert_remapping_progress_to_db(private_config_xml_file, dataframe):
+    list_to_remap = dataframe.values.tolist()
+    if len(list_to_remap) > 0:
+        with psycopg2.connect(get_pg_metadata_uri_for_eva_profile("development", private_config_xml_file),
+                              user="evadev") as metadata_connection_handle:
+            with metadata_connection_handle.cursor() as cursor:
+                query_insert = (
+                    'INSERT INTO remapping_progress '
+                    '(source, taxid, scientific_name, assembly_accession, number_of_study, '
+                    'number_submitted_variants, target_assembly_accession, release_number) '
+                    'VALUES %s'
+                )
+                psycopg2.extras.execute_values(cursor, query_insert, list_to_remap)
+
+
+def main():
+    argparse = ArgumentParser(
+        description='Gather the current set of studies from both EVA and dbSNP that can be remapped, clustered '
+                    'and released. The source of EVA studies is the metadata database and the source of dbSNP studies '
+                    "is last year's spreadsheet. The number of variants are populated from counts retrieved from "
+                    ' eva_stats.')
+    argparse.add_argument('--input', help='Path to the file containing the taxonomies and assemblies', required=True)
+    argparse.add_argument('--output', help='Path to the file that will contain the input plus annotation',
+                          required=True)
+    argparse.add_argument('--private_config_xml_file', required=True,
+                          help='Path to the file containing the username/passwords tp access '
+                               'production and development databases')
+    args = argparse.parse_args()
+    output_header = ['Source', 'Taxid', 'Scientific Name', 'Assembly', 'number Of Studies',
+                     'Number Of Variants (submitted variants)', 'Ensembl assembly from taxid',
+                     'Ensembl assembly from assembly', 'Target Assembly']
+
+    accession_counts = get_accession_counts_per_assembly(args.private_config_xml_file, 'dbSNP')
+    df1 = parse_dbsnp_csv(args.input, accession_counts)
+    accession_counts = get_accession_counts_per_study(args.private_config_xml_file, 'EVA')
```

Comment:
Suggest not reusing the accession_counts variable for clarity..

Reply:
Done